plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -200,18 +214,68 @@ def sweep(self, arguments: List[str]) -> None:
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
             self.job_idx += len(returns)
             for trial, ret in zip(trials, returns):
-                study._tell(trial, optuna.trial.TrialState.COMPLETE, [ret.return_value])
+                if len(directions) == 1:
+                    try:
+                        study._tell(
+                            trial,
+                            optuna.trial.TrialState.COMPLETE,
+                            [float(ret.return_value)],
+                        )
+                    except (TypeError, ValueError):
+                        study._tell(trial, optuna.trial.TrialState.FAIL, None)
+                        log.warning(
+                            f"Return value must be a float value. Got {ret.return_value}."
+                        )
+                else:
+                    if not isinstance(ret.return_value, (list, tuple)):
+                        study._tell(trial, optuna.trial.TrialState.FAIL, None)
+                        log.warning(
+                            f"Return value is not a sequence. Got {ret.return_value}."
+                        )
```

Comment:
Do we need to return early here or raise an Error? Or the application will error out itself?

Reply:
Thank you for pointing it out. We need to raise errors because the user code should be fixed to continue optimization correctly. I applied your suggestion in the commit https://github.com/facebookresearch/hydra/pull/1294/commits/78b6f351fffbefe328f26653c392618cb94edc8f.